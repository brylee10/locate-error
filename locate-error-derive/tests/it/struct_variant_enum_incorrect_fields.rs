//! Test invalid enums with struct variants

use locate_error::Locate;
use locate_error::Location;

// Too many fields
#[derive(Locate)]
enum TestEnum {
    Variant1 {
        #[locate_from]
        source: String,
        location: Location,
        extra: String,
    },
}

// No location field
#[derive(Locate)]
enum TestEnum2 {
    Variant1 {
        #[locate_from]
        source: String,
        // Missing location field
    },
}

// No locate_from attribute
#[derive(Locate)]
enum TestEnum3 {
    Variant1 {
        source: String, // Missing #[locate_from]
        location: Location,
    },
    Variant2 {
        source: String,
    },
}

// Too many fields with locate_from attribute
#[derive(Locate)]
enum TestEnum4 {
    Variant1 {
        #[locate_from]
        source: String,
        location: Location,
        #[locate_from]
        extra: u32,
    },
}

// No location field
#[derive(Locate)]
enum TestEnum5 {
    Variant1 {
        #[locate_from]
        source: String,
        value: u32, // This should be a location field
    },
}

fn main() {}
