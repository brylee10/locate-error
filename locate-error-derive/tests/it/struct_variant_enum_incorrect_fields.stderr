error: Locate requires enums variants with the #[locate_from] attribute to have exactly two fields, one for the source and one for the location
 --> tests/it/struct_variant_enum_incorrect_fields.rs:9:5
  |
9 |     Variant1 {
  |     ^^^^^^^^

error: Variants with #[locate_from] must have a field named 'location' of type `locate_from::Location`
  --> tests/it/struct_variant_enum_incorrect_fields.rs:20:5
   |
20 |     Variant1 {
   |     ^^^^^^^^

error: Locate requires at least one variant with the #[locate_from] attribute (otherwise this macro is effectively a no-op)
  --> tests/it/struct_variant_enum_incorrect_fields.rs:28:10
   |
28 | #[derive(Locate)]
   |          ^^^^^^
   |
   = note: this error originates in the derive macro `Locate` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Locate requires enums variants with the #[locate_from] attribute to have exactly two fields, one for the source and one for the location
  --> tests/it/struct_variant_enum_incorrect_fields.rs:42:5
   |
42 |     Variant1 {
   |     ^^^^^^^^

error: Variants with #[locate_from] must have a field named 'location' of type `locate_from::Location`
  --> tests/it/struct_variant_enum_incorrect_fields.rs:54:5
   |
54 |     Variant1 {
   |     ^^^^^^^^
